### ISSUE TITLE
SUSE Linux/Tumbleweed build doese not work

### ISSUE THREAD

**Issue by LowLevelMahn** [open-watcom/open-watcom-v2#1339]
latest Tumbleweed with gcc 14.2

1. git clone https://github.com/open-watcom/open-watcom-v2.git

(didn't copied the setvars.sh)

```
diff --git a/setvars.sh b/setvars.sh
index d011833882..cfb5ad7024 100755
--- a/setvars.sh
+++ b/setvars.sh
@@ -34,12 +34,12 @@ export OWDOCBUILD=0
 # It doesn't requires appropriate DOSBOX emulator
 # and suppress documentation build

-# export OWNOWGML=1
+export OWNOWGML=1

 # Set OWGUINOBUILD variable to 1 to suppress tools GUI version build
 # If it is used then only tools character mode version is build

-# export OWGUINOBUILD=1
+export OWGUINOBUILD=1

 # Set OWNOBUILD variable to list of OW projects to suppress their build
 # Example export OWNOBUILD=ide browser dlgprs
@@ -72,7 +72,7 @@ export OWDISTRBUILD=0
 # if DOSBOX emulator is used then OWDOSBOX variable must be set
 # Uncoment and set OWDOSBOX variable bellow to point to DOSBOX emulator executable

-# export OWDOSBOX=dosbox
+export OWDOSBOX=dosbox
```
2. ./setvars.sh
3. ./build.sh 

returns

```
Open Watcom build environment (GCC version=14)
wmake bootstrap build error

```

**Comment by jmalak** [comment]
please check build/binbuild/bootx.log for messages. 
Probably gcc version 14 generate some new messages and need to fix them.

**Comment by LowLevelMahn** [comment]
here the content of the bootx.log

```
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2/build/binbuild> cat bootx.log
rm -f *.o *.gh *.exe
c99 -g -O2 -I. -I../h -I../../watcom/h -I../../lib_misc/h -D__LINUX__ -D__UNIX__ -DBOOTSTRAP -DINCL_MSGTEXT -D_POSIX_C_SOURCE=200112L -D_BLDVER=1300 -D_CYEAR=xxxx -c ../../builder/c/wsplice.c
c99: invalid option -- '2'
usage: c99 [-c] [-D name[=value]] [...] [-E] [-g] [-I directory ...]
       [-L directory ...] [-o outfile] [-O] [-s] [-U name ...] operand ...

       where operand is one or more of file.c, file.o, file.a
       or -llibrary
make: *** [../posmake:69: wsplice.o] Error 1
```

**Comment by jmalak** [comment]
It looks like change or bug in c99 for your installation.
The problem is with -O2 option of full command

```
c99 -g -O2 -I. -I../h -I../../watcom/h -I../../lib_misc/h -D__LINUX__ -D__UNIX__ -DBOOTSTRAP -DINCL_MSGTEXT -D_POSIX_C_SOURCE=200112L -D_BLDVER=1300 -D_CYEAR=xxxx -c ../../builder/c/wsplice.c
```

**Comment by LowLevelMahn** [comment]
i will ask on the gcc mailing list

**Comment by jmalak** [comment]
I tried it with gcc 12 and no problem with -O2 option.
I don't know what shell you use, OW build system suppose POSIX compatible Korn/Bash shell.

**Comment by LowLevelMahn** [comment]
> I don't know what shell you use, OW build system suppose POSIX compatible Korn/Bash shell.

```
echo "$SHELL"
/bin/bash

bash --version
GNU bash, version 5.2.37(1)-release (x86_64-suse-linux)
```

**Comment by LowLevelMahn** [comment]
on Ubuntu
the /usr/bin/c99 is a link to an script that just calls gcc

on Tumbleweed its some (c compiler related) binary with no link to gcc - even the manual looks different to Ubuntu/Debian

**Comment by jmalak** [comment]
It doesn't matter, c99 command line is defined by POSIX standard.

```
-O  optlevel
Specify the level of code optimization. If the optlevel option-argument is the digit '0', all special code 
optimizations shall be disabled. If it is the digit '1', the nature of the optimization is unspecified. 
If the -O option is omitted, the nature of the system's default optimization is unspecified. It is unspecified
whether code generated in the presence of the -O 0 option is the same as that generated when -O is omitted.
Other optlevel values may be supported.
```
that optlevel value can be toolchain specific.

**Comment by LowLevelMahn** [comment]
its a bug in Tumbleweed - c99 is not correctly targeting gcc but some ancient c compiler (if i get it right) - its seems to be known

**Comment by jmalak** [comment]
It is not possible because we relate on POSIX c99 not on gcc in pre-boot phase for *NIX systems (by example it uses c99 on OSX etc.
This pre-boot phase must be portable and cannot depend on any specific compiler therefore we use POSIX c99.
I can do exception for gcc but I am not sure how gcc command line is different from c99 syntax.

**Comment by LowLevelMahn** [comment]
> It is not possible because we relate on POSIX c99 not on gcc in pre-boot

on Ubuntu 24.04.1 LTS it seems that c99 is just a shell script that calls gcc

`/usr/bin/c99 -> /etc/alternatives/c99`

```
cat /etc/alternatives/c99
#! /bin/sh

# Call the appropriate C compiler with options to accept ANSI/ISO C
# The following options are the same (as of gcc-3.3):
# 	-std=c99
# 	-std=c9x
# 	-std=iso9899:1999
# 	-std=iso9899:199x

extra_flag=-std=c99

for i; do
    case "$i" in
	-std=c9[9x]|-std=iso9899:199[9x])
	    extra_flag=
	    ;;
	-std=*|-ansi)
	    echo >&2 "`basename $0` called with non ISO C99 option $i"
	    exit 1
	    ;;
    esac
done

exec gcc $extra_flag ${1+"$@"}

```

**Comment by LowLevelMahn** [comment]
the build Fails on Fedora 40 gcc 14.2.1 and clang 18.1.8 the same way

```
...
./mkcode.exe copystub.gh
bld_as decstub.obj
bld_link decstub.tmp
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
./mkcode.exe decstub.gh
cc cwc.obj
cl cwc.exe
/usr/bin/ld: cannot find -lc: No such file or directory
collect2: error: ld returned 1 exit status
Error(E42): Last command making (cwc.exe) returned a bad status
Error(E02): Make execution terminated
<pmake -d build          -h> => non-zero return: 512
Build failed

```

**Comment by jmalak** [comment]
It will need to modify bld/wmake/posmake (POSIX make file) to use gcc command instead of c99 if gcc toolchain is used.
I will try to change and test.
The **`-lc`** option should be option to include standard C library. try to go to **`bld/causeway/cwc`** and run **`builder clean`** and next **`builder build OWVERBOSE=1`** to get full command line.

**Comment by LowLevelMahn** [comment]
```
test@fedora:~/temp/open-watcom-v2_dev/open-watcom-v2_gcc$ cd bld/causeway
test@fedora:~/temp/open-watcom-v2_dev/open-watcom-v2_gcc/bld/causeway$ builder clean
bash: builder: command not found...
test@fedora:~/temp/open-watcom-v2_dev/open-watcom-v2_gcc/bld/causeway$ ./builder clean
bash: ./builder: No such file or directory

```

**Comment by jmalak** [comment]
It suppose you have setup OW build environment and have run OW build up to break.
Anyway I did workaround fix for you, add following code to your build.sh script to fix issue with c99.

```
if [ "$OWTOOLS" = "WATCOM" ]; then
    output_redirect wmake -m -f ../wmake clean
    output_redirect wmake -m -f ../wmake
else
    if [ "$OWTOOLS" = "GCC" ]; then
        if [ "$OWTOOLSVER" = "14" ]; then
            # workaround for gcc 14 on SUSE
            TCC="CC=gcc"
        fi 
    fi
    case `uname` in
        FreeBSD)
            output_redirect make -f ../posmake clean
            output_redirect make -f ../posmake TARGETDEF=-D__BSD__ $TCC
            ;;
        Darwin)
            output_redirect make -f ../posmake clean
            output_redirect make -f ../posmake TARGETDEF=-D__OSX__ $TCC
            ;;
        Haiku)
            output_redirect make -f ../posmake clean
            output_redirect make -f ../posmake TARGETDEF=-D__HAIKU__ $TCC
            ;;
#        Linux)
        *)
            output_redirect make -f ../posmake clean
            output_redirect make -f ../posmake TARGETDEF=-D__LINUX__ $TCC
            ;;
    esac
fi
```

**Comment by LowLevelMahn** [comment]
> It suppose you have setup OW build environment and have run OW build up to break.

i try that

> Anyway I did workaround fix for you, modify your build.sh to be as following code to fix issue with c99 and gcc 14.

compilations starts and seems to work

**Comment by LowLevelMahn** [comment]
i got a compilation error on Ubuntu 24.10 (also with gcc 14.2.x)

```
cc guixloop.obj
../c/guixloop.c: In function ‘GUICreatePopup’:
../c/guixloop.c:227:38: error: passing argument 2 of ‘GUICreateMenuPopup’ from incompatible pointer type [-Wincompatible-pointer-types]
  227 |     ui_ev = GUICreateMenuPopup( wnd, &scr_location, wnd->menu, GUI_TRACK_LEFT, NULL );
      |                                      ^~~~~~~~~~~~~
      |                                      |
      |                                      guix_coord *
In file included from ../c/guixloop.c:40:
../../../../bld/gui/ui/h/guifloat.h:34:66: note: expected ‘guix_point *’ but argument is of type ‘guix_coord *’
   34 | extern ui_event GUICreateMenuPopup( gui_window *wnd, guix_point *scr_location, UIMENUITEM *menuitems,
      |                                                      ~~~~~~~~~~~~^~~~~~~~~~~~
Error(E42): Last command making (guixloop.obj) returned a bad status
Error(E02): Make execution terminated
<pmake -d build          -h> => non-zero return: 512
Build failed

```

**Comment by LowLevelMahn** [comment]
don't be confused: i try to build openwatcom on recent Tumbleweed/Ubuntu 24.10 and Fedora 40 in parallel :)

**Comment by LowLevelMahn** [comment]
> > It suppose you have setup OW build environment and have run OW build up to break.
> 
> i try that

i did run my ./setvar.sh and ./build.sh up to break but the builder command don't get found afterwards?

**Comment by jmalak** [comment]
Run **`sh build.sh preboot`** and check in **`build/binbuild`** if there is **`wmake and builder`** compiled executable.

**Comment by jmalak** [comment]
May be a problem that you use template script **`setvars.sh`** instead of your **`own configured script`**.

**Comment by LowLevelMahn** [comment]
> Run **`sh build.sh preboot`** and check in **`build/binbuild`** if there is **`wmake and builder`** compiled executable.

```
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2> sh build.sh preboot
Open Watcom build environment (GCC version=14)
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2> ls build/binbuild/
aliasgen  bmp2eps    bspecs.owc  bwasm    bwcc386  bwcl386  bwhc    bwpp386  chk_exec  cvtutf8   genverrc   mkinf     objxdef   parsedyn   uzip    wsplice
bcwc      boot.lo1   builder     bwasmps  bwccaxp  bwclaxp  bwipfc  bwppaxp  chktest   dlgprs    hcdos      msgencod  objxref   re2c       vicomp  yacc
bdmpobj   boot.log   build.lo1   bwasppc  bwccmps  bwclmps  bwlib   bwrc     comstrip  errdoc    kanji.uni  objchg    optencod  ssl        wcpp
bedbind   bootx.log  build.log   bwbind   bwccppc  bwclppc  bwlink  bwstrip  cpyright  errsrc    langdat    objfind   parsectl  sweep      whpcvt
bide2mak  bowcc      bwasaxp     bwcc     bwcl     bwdis    bwpp    catfile  crlf      findhash  mkexezip   objlist   parsedlg  to932.uni  wmake

```

both available

**Comment by jmalak** [comment]
for sure run **`sh clean.sh`** before `sh build.sh ...` to cleanup all residual build objects.

**Comment by LowLevelMahn** [comment]
> May be a problem that you use template script **`setvars.sh`** instead of your **`own configured script`**.

i will fresh re-clone, copy the setvars.sh into my own, apply your c99 patch and retry

**Comment by LowLevelMahn** [comment]
> for sure run **`sh clean.sh`** before `sh build.sh ...` to cleanup all residual build objects.

did that - work with just ./clean.sh

**Comment by LowLevelMahn** [comment]
> May be a problem that you use template script **`setvars.sh`** instead of your **`own configured script`** created from this template.

i changed that for the freshly started Tumbleweed and Fedora build - Ubuntu seems to get further without doing that... but i'll give it a try

**Comment by LowLevelMahn** [comment]
the freshly started Tumbleweed build fails as before - fresh git clone/build.sh c99 patch at the same location

```
...
bld_as decstub.obj
bld_link decstub.tmp
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
./mkcode.exe decstub.gh
cc cwc.obj
cl cwc.exe
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: cannot find -lc: No such file or directory
collect2: error: ld returned 1 exit status
Error(E42): Last command making (cwc.exe) returned a bad status
Error(E02): Make execution terminated
<pmake -d build          -h> => non-zero return: 512
Build failed
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2>
```

im sitting right there

```
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2> ls build/binbuild/ | grep 'builder\|wmake'
builder
wmake
```

```
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2> cd bld/causeway/
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway> builder clean
-bash: builder: command not found
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway>
```
but builder seems not findable due to my path environment - how should that work?

**Comment by jmalak** [comment]
it looks like you have incorrectly setup OW build environment.
You need to run your setup script before you work in build environment.
It setup PATH to contains OW build binaries, you don't have it setup that you do something wrong (not run setup script).
Setup script must be run  by **`. <setup script>`** to hold changed environment.

**Comment by LowLevelMahn** [comment]
> it looks like you have incorrectly setup OW build environment.
> You need to run your setup script before you work in build environment.

where is this setup-script? im calling my copy of setvars.sh ../my_setvars.sh (with correctly set OWROOT) and then ./build.sh
is there some other script? can't find that clearly in the docs

**Comment by jmalak** [comment]
no, only this one but you must run it by **`.  ./my_setvars.sh`** to hold changes but you run it as  **`./my_setvars.sh`** what is wrong.

**Comment by LowLevelMahn** [comment]
> no, only this one but you must run it by **`. ./my_setvars.sh`** to hold changes but you run it as **`./my_setvars.sh`** what is wrong.

i missed that detail - sorry

**Comment by LowLevelMahn** [comment]
the error:

```
= 11:09:11 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/linux386 =
= 11:09:11 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/linuxx64 =
cl cwc.exe
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: cannot find -lc: No such file or directory
collect2: error: ld returned 1 exit status
Error(E42): Last command making (cwc.exe) returned a bad status
Error(E02): Make execution terminated
<pmake -d build guitool .not .and         -h> => non-zero return: 512
Build failed

```

the cleanup

```
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2> cd bld
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2/bld> cd causeway/
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway> builder clean
**** CLEAN rule
== 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway ===
= 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cw32/dos386 ==
rm -f *.obj
rm -f *.lnd *.lnk *.exe *.dll *.lib *.lbc *.res *.lst *.err *.pch *.map *.sym *.tmp *.trk *.exe
= 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/dosi86 =
rm -f *.obj  *.gh
rm -f *.lnd *.lnk *.exe *.dll *.lib *.lbc *.res *.lst *.err *.pch *.map *.sym *.tmp *.trk
= 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/linux386 =
rm -f *.obj  *.gh
rm -f *.lnd *.lnk *.exe *.dll *.lib *.lbc *.res *.lst *.err *.pch *.map *.sym *.tmp *.trk
= 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/linuxx64 =
rm -f *.obj  *.gh
rm -f *.lnd *.lnk *.exe *.dll *.lib *.lbc *.res *.lst *.err *.pch *.map *.sym *.tmp *.trk
= 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/nt386 ==
rm -f *.obj  *.gh
rm -f *.lnd *.lnk *.exe *.dll *.lib *.lbc *.res *.lst *.err *.pch *.map *.sym *.tmp *.trk
= 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/ntx64 ==
rm -f *.obj  *.gh
rm -f *.lnd *.lnk *.exe *.dll *.lib *.lbc *.res *.lst *.err *.pch *.map *.sym *.tmp *.trk
= 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/os2386 =
rm -f *.obj  *.gh
rm -f *.lnd *.lnk *.exe *.dll *.lib *.lbc *.res *.lst *.err *.pch *.map *.sym *.tmp *.trk
= 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwdll/dos386 =
rm -f *.obj
rm -f *.lnd *.lnk *.exe *.dll *.lib *.lbc *.res *.lst *.err *.pch *.map *.sym *.tmp *.trk
== 11:09:24 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway ===
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway> cd ..
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2/bld> cd causeway/
```

build verbose

```
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway> builder build OWVERBOSE=1
**** BUILD rule
== 11:09:55 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway ===
= 11:09:56 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cw32/dos386 ==
bwasm -q  -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -we -d1 -fo=cw32.obj -w8-cx -bt=dos  -DENGLISH=1 -I"../../inc"      -I"../../../bld/hdr/dos/h"   -I"../../../bld/watcom/h" ../../asm/cw32.asm
../../asm/raw_vcpi.asm(359): Note! N303: Far call is converted to near call.
../../asm/raw_vcpi.asm(359): Note! N290: included by file ../../asm/cw32.asm(740)
../../asm/raw_vcpi.asm(364): Note! N303: Far call is converted to near call.
../../asm/raw_vcpi.asm(364): Note! N290: included by file ../../asm/cw32.asm(740)
../../asm/raw_vcpi.asm(369): Note! N303: Far call is converted to near call.
../../asm/raw_vcpi.asm(369): Note! N290: included by file ../../asm/cw32.asm(740)
../../asm/raw_vcpi.asm(374): Note! N303: Far call is converted to near call.
../../asm/raw_vcpi.asm(374): Note! N290: included by file ../../asm/cw32.asm(740)
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
../../asm/cw32.asm(941): Note! N293: Extending jump
../../asm/cw32.asm(956): Note! N293: Extending jump
../../asm/cw32.asm(975): Note! N293: Extending jump
../../asm/cw32.asm(4511): Note! N293: Extending jump
../../asm/cw32.asm(4520): Note! N293: Extending jump
../../asm/cw32.asm(4522): Note! N293: Extending jump
../../asm/cw32.asm(4524): Note! N293: Extending jump
%write cw32.lnk debug dwarf op nored op symfile op map sys dos    op verbose, statics disable 1080
bwlink op q name cw32.exe @cw32.lnk
bcwc cw32.exe cwstub.exe
bwasm -q  -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -we -d1 -fo=cwdstub.obj -w8-cx -bt=dos  -DENGLISH=1 -I"../../inc"      -I"../../../bld/hdr/dos/h"   -I"../../../bld/watcom/h" ../../asm/cwdstub.asm
%write cwdstub.lnk debug dwarf op nored op symfile op map sys dos    op verbose, statics disable 1014
bwlink op q name cwdstub.exe @cwdstub.lnk
= 11:09:56 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/dosi86 =
gcc -pipe -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG -D_ENABLE_AUTODEPEND -D__UNIX__ -D__FLAT__ -D__LINUX__ -D_M_X64 -D__LINUX_X64__ ../../../../bld/watcom/c/mkcode.c -fno-asm -fno-common -fsigned-char -Wall -Wno-switch -Wno-missing-braces -Wno-parentheses -O -g1 -gdwarf-2 -gz  -I"../../../../bld/watcom/h"  -o ./mkcode.exe  -Wl,-Map,mkcode.map ../../../../bld/watcom/binbuild/clibext.lib
bwasm -zq-w8-cx -fo=copystub.obj ../../asm/copystub.asm
bwlink system bin op quiet file copystub.obj name copystub.tmp
./mkcode.exe -b copystub.tmp copystub.gh
bwasm -zq-w8-cx -fo=decstub.obj ../../asm/decstub.asm
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
bwlink system bin op quiet file decstub.obj name decstub.tmp
./mkcode.exe -b decstub.tmp decstub.gh
bwcc -zq  -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -D_ENABLE_AUTODEPEND          -j-we -oaxt -d1 -x-wpx-xx-zam -fo=cwc.obj -fpi -mc -bt=dos  -s     -wx-wce=C310 -zastd=c99  -I.  -I"../../../../bld/hdr/dos/h"   -I"../../../../bld/watcom/h" ../../c/cwc.c
%write cwc.lnk debug dwarf op nored op symfile op map sys dos
bwlink op q name cwc.exe @cwc.lnk
= 11:09:56 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/linux386 =
gcc -pipe -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG -D_ENABLE_AUTODEPEND -D__UNIX__ -D__FLAT__ -D__LINUX__ -D_M_X64 -D__LINUX_X64__ ../../../../bld/watcom/c/mkcode.c -fno-asm -fno-common -fsigned-char -Wall -Wno-switch -Wno-missing-braces -Wno-parentheses -O -g1 -gdwarf-2 -gz  -I"../../../../bld/watcom/h"  -o ./mkcode.exe  -Wl,-Map,mkcode.map ../../../../bld/watcom/binbuild/clibext.lib
bwasm -zq-w8-cx -fo=copystub.obj ../../asm/copystub.asm
bwlink system bin op quiet file copystub.obj name copystub.tmp
./mkcode.exe -b copystub.tmp copystub.gh
bwasm -zq-w8-cx -fo=decstub.obj ../../asm/decstub.asm
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
bwlink system bin op quiet file decstub.obj name decstub.tmp
./mkcode.exe -b decstub.tmp decstub.gh
bwcc386 -zq  -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -D_ENABLE_AUTODEPEND         -j-we -oaxt -d1 -x-wpx-xx-zam -fo=cwc.obj -fpi -mf -zc -bt=linux  -s     -wx-wce=C310 -zastd=c99  -I.  -I"../../../../bld/hdr/linux/h"   -I"../../../../bld/watcom/h" ../../c/cwc.c
%write cwc.lnk debug dwarf op nored op symfile op map sys linux
bwlink op q name cwc.exe @cwc.lnk
= 11:09:56 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/causeway/cwc/linuxx64 =
gcc -pipe -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -D__UNIX__ -D__FLAT__ -D__LINUX__ -D_M_X64 -D__LINUX_X64__ ../../../../bld/watcom/c/mkcode.c -fno-asm -fno-common -fsigned-char -Wall -Wno-switch -Wno-missing-braces -Wno-parentheses -O -g1 -gdwarf-2 -gz  -I"../../../../bld/watcom/h"  -o ./mkcode.exe  -Wl,-Map,mkcode.map ../../../../bld/watcom/binbuild/clibext.lib
bwasm -zq-w8-cx -fo=copystub.obj ../../asm/copystub.asm
bwlink system bin op quiet file copystub.obj name copystub.tmp
./mkcode.exe -b copystub.tmp copystub.gh
bwasm -zq-w8-cx -fo=decstub.obj ../../asm/decstub.asm
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
../../asm/decstub.asm(268): Note! N293: Extending jump
../../asm/decstub.asm(285): Note! N293: Extending jump
bwlink system bin op quiet file decstub.obj name decstub.tmp
./mkcode.exe -b decstub.tmp decstub.gh
gcc -pipe -c -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG   -D__UNIX__ -D__FLAT__ -D_M_X64 -D__LINUX_X64__ -D__LINUX__      -fno-asm -fno-common -fsigned-char -Wall -Wno-switch -Wno-missing-braces -Wno-parentheses  -O -g1 -gdwarf-2 -gz -static -o cwc.obj -std=gnu99  -I.   -I"../../../../bld/watcom/h" ../../c/cwc.c
gcc -pipe -static  -O -g1 -gdwarf-2 -gz -o cwc.exe  -Wl,-Map,cwc.map cwc.obj   ../../../../bld/watcom/linuxx64/clibext.lib/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: cannot find -lc: No such file or directory
collect2: error: ld returned 1 exit status
Error(E42): Last command making (cwc.exe) returned a bad status
Error(E02): Make execution terminated
<pmake -d build guitool .not .and         -h> => non-zero return: 512
Build failed
```

**Comment by jmalak** [comment]
It looks like you don't have install static C run-time library on your system. Maybe that now with gcc only shared libraries are installed not static library or it is gcc installation bug. But I am not expert for Linux and gcc what and how it should be install.
OW build uses static C run-time library.

**Comment by jmalak** [comment]
The gui type mismatch is fixed now in git repository.

**Comment by LowLevelMahn** [comment]
> It looks like you don't have install static C run-time library on your system. Maybe that now with gcc only shared libraries are installed not static library or it is gcc installation bug. But I am not expert for Linux and gcc what and how it should be install. OW build uses static C run-time library that must be on system.

SUSE and Fedora aren't that unusal - with Ubuntu the big three in the linux world
i wil check if the static libs a installed by default on this distros

**Comment by LowLevelMahn** [comment]
but Tumbleweed and Fedora still fail differently

Tumbleweed:

```
cd bld/aui
builder clean and 
builder build OWVERBOSE=1
```

```
I"../../../bld/aui/nt386" -I"../../../bld/aui/h" -I"../../../bld/gui/ui/nt386" -I"../../../bld/gui/h" -I"../../../bld/ui/h"  -I"../../../bld/hdr/dos/h" -I"../../../bld/w32api/nt/h"  -I"../../../bld/watcom/h" ../c/wndmnsz.c
bwcc386 -zq  -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -D_ENABLE_AUTODEPEND   -DWINVER=0x400 -D_WIN32_IE=0x300 -DWIN32_LEAN_AND_MEAN      -j-we -oaxt -d1 -x-wpx-xx-zam -fo=wndimain.obj -fpi -mf -bt=nt  -s     -wx-wce=C310 -zastd=c99  -I. -I"../h" -I"../../../bld/aui/nt386" -I"../../../bld/aui/h" -I"../../../bld/gui/ui/nt386" -I"../../../bld/gui/h" -I"../../../bld/ui/h"  -I"../../../bld/hdr/dos/h" -I"../../../bld/w32api/nt/h"  -I"../../../bld/watcom/h" ../c/wndimain.c
bwlib -q -b-c-n-pa  -ii  -io   aui.lib  guimain.obj wndgadgt.obj wndrect.obj wndtbatr.obj wndbgch.obj wndstyle.obj wndhist.obj wndhelp.obj wndstat.obj wndchoos.obj wndcreat.obj wndcrsr.obj wndcurr.obj wndid.obj wndmem.obj wndmeth.obj wndutil.obj wndchar.obj wndpop.obj rxwrap.obj wndscrol.obj wndsel.obj wndsrch.obj wndfdlg.obj wnddlg.obj dlgsrch.obj dlgpick.obj dlgnew.obj wndspawn.obj wndalloc.obj wndshrnk.obj wndtool.obj wnddoin.obj wndresiz.obj wndsysin.obj wndclean.obj wndmacro.obj wndring.obj wndefra.obj wndfonhk.obj wndshthk.obj wndqryhk.obj wndendhk.obj wndrszhk.obj wndsfra.obj wndfont.obj wndldstr.obj wndfront.obj wndpaint.obj wnddirt.obj wnddfclr.obj wndcolor.obj wndmouse.obj wndrxerr.obj wndfref.obj wndmsg.obj wndmnsz.obj wndimain.obj
= 12:49:22 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/aui/nt386.gui =
awk -v prefix=AUI_LITERAL_ -v base=AUI_LITERAL_BASE -f ../../../build/makemsg.awk -v OUTFILE=aui.grh ../res/aui.msg
bwcc386 -zq  -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -D_ENABLE_AUTODEPEND   -DWINVER=0x400 -D_WIN32_IE=0x300 -DWIN32_LEAN_AND_MEAN      -j-we -oaxt -d1 -x-wpx-xx-zam -fo=guimain.obj -fpi -mf -bt=nt  -s      -wx-wce=C310 -zastd=c99  -I. -I"../h" -I"../../../bld/aui/nt386.gui" -I"../../../bld/aui/h" -I"../../../bld/gui/win/nt386" -I"../../../bld/gui/h"   -I"../../../bld/hdr/dos/h" -I"../../../bld/w32api/nt/h"  -I"../../../bld/watcom/h" ../c/guimain.c
../../../bld/gui/h/gui.rh(44): Error! E1055: Unable to open 'guimsgs.grh'
Error(E42): Last command making (guimain.obj) returned a bad status
Error(E02): Make execution terminated
<pmake -d build guitool .not .and         -h> => non-zero return: 512
Build failed
```

**Comment by jmalak** [comment]
It looks like DOSBOX issue. 
DOSBOX is configured without sound and display to work in batch only.
Maybe some changes in DOSBOX behaviour, it looks like it requires display.
Probably we will need to change configuration of DOSBOX to use display.

**Comment by LowLevelMahn** [comment]
> Try to setup OWWGMLDEBUG=1 in environment, it should switch on DOSBOX GUI box and you could get what exactly happen.
> Anyway I correct null device definition for dosbox, but I don't think it is related to your issue.

something changed with Fedora after pulling your change
and with installing the missing static libstdc++ with `sudo dnf install libstdc++-static` the build ended without errors

Tumbleweed still struggles with some (but different to Fedora) error

```
cd bld/aui
builder clean and 
builder build OWVERBOSE=1
```

```
bwcc386 -zq  -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -D_ENABLE_AUTODEPEND   -DWINVER=0x400 -D_WIN32_IE=0x300 -DWIN32_LEAN_AND_MEAN      -j-we -oaxt -d1 -x-wpx-xx-zam -fo=wndimain.obj -fpi -mf -bt=nt  -s     -wx-wce=C310 -zastd=c99  -I. -I"../h" -I"../../../bld/aui/nt386" -I"../../../bld/aui/h" -I"../../../bld/gui/ui/nt386" -I"../../../bld/gui/h" -I"../../../bld/ui/h"  -I"../../../bld/hdr/dos/h" -I"../../../bld/w32api/nt/h"  -I"../../../bld/watcom/h" ../c/wndimain.c
bwlib -q -b-c-n-pa  -ii  -io   aui.lib  guimain.obj wndgadgt.obj wndrect.obj wndtbatr.obj wndbgch.obj wndstyle.obj wndhist.obj wndhelp.obj wndstat.obj wndchoos.obj wndcreat.obj wndcrsr.obj wndcurr.obj wndid.obj wndmem.obj wndmeth.obj wndutil.obj wndchar.obj wndpop.obj rxwrap.obj wndscrol.obj wndsel.obj wndsrch.obj wndfdlg.obj wnddlg.obj dlgsrch.obj dlgpick.obj dlgnew.obj wndspawn.obj wndalloc.obj wndshrnk.obj wndtool.obj wnddoin.obj wndresiz.obj wndsysin.obj wndclean.obj wndmacro.obj wndring.obj wndefra.obj wndfonhk.obj wndshthk.obj wndqryhk.obj wndendhk.obj wndrszhk.obj wndsfra.obj wndfont.obj wndldstr.obj wndfront.obj wndpaint.obj wnddirt.obj wnddfclr.obj wndcolor.obj wndmouse.obj wndrxerr.obj wndfref.obj wndmsg.obj wndmnsz.obj wndimain.obj
= 16:03:17 /home/linux/temp/open-watcom-v2_dev/open-watcom-v2/bld/aui/nt386.gui =
awk -v prefix=AUI_LITERAL_ -v base=AUI_LITERAL_BASE -f ../../../build/makemsg.awk -v OUTFILE=aui.grh ../res/aui.msg
bwcc386 -zq  -D_BLDVER=1300 -D_CYEAR=2024 -DNDEBUG  -D_ENABLE_AUTODEPEND   -DWINVER=0x400 -D_WIN32_IE=0x300 -DWIN32_LEAN_AND_MEAN      -j-we -oaxt -d1 -x-wpx-xx-zam -fo=guimain.obj -fpi -mf -bt=nt  -s      -wx-wce=C310 -zastd=c99  -I. -I"../h" -I"../../../bld/aui/nt386.gui" -I"../../../bld/aui/h" -I"../../../bld/gui/win/nt386" -I"../../../bld/gui/h"   -I"../../../bld/hdr/dos/h" -I"../../../bld/w32api/nt/h"  -I"../../../bld/watcom/h" ../c/guimain.c
../../../bld/gui/h/gui.rh(44): Error! E1055: Unable to open 'guimsgs.grh'
Error(E42): Last command making (guimain.obj) returned a bad status
Error(E02): Make execution terminated
<pmake -d build guitool .not .and         -h> => non-zero return: 512
Build failed
```

**Comment by jmalak** [comment]
The issue is with gui project because guimsgs.grh is created by gui project.
First rebuild gui project and next aui project, this projects are dependent. For gui project ui project is prerequsity etc.
anyway guimsgs.grh is created by awk script. It could be issue if Tumbleweed's awk is not standard.
check gui project if guimsgs.grh is created for all targets.

**Comment by LowLevelMahn** [comment]
```
cd bld/gui
builder clean and 
builder build OWVERBOSE=1

```
i can't see any error in the output

and these are the files i've found (searching from ow-root)

```
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2> find . -name "guimsgs.grh"
./bld/gui/ui/rdos386/guimsgs.grh
./bld/gui/ui/nt386/guimsgs.grh
./bld/gui/ui/os2386/guimsgs.grh
./bld/gui/ui/dosi86/guimsgs.grh
./bld/gui/ui/linux386/guimsgs.grh
./bld/gui/ui/dos386/guimsgs.grh
./bld/gui/ui/wini86/guimsgs.grh
./bld/gui/ui/linuxx64/guimsgs.grh
```

log from gui OWVERBOSE=1 built are just to big to post here or on pastebin.com - but the logs contain no error or warning

awk Version is the same 5.3.0 on Fedora(build ok),Tumbleweed(build fails) and Ubuntu(build ok)

**Comment by jmalak** [comment]
It is requesting **`bld/gui/win/nt386/guimsgs.grh`** that you need check this one.
for **`nt386`** exists two versions 

**`bld/gui/ui/nt386/guimsgs.grh`** for ui character mode 

**`bld/gui/win/nt386/guimsgs.grh`** for win GUI mode

**Comment by jmalak** [comment]
There are two versions GUI and UI.
You check only UI version which compile without problem, but GUI version  is failing.
Check **bld/gui/win/nt386/guimsgs.grh**

**Comment by LowLevelMahn** [comment]
> Check **bld/gui/win/nt386/guimsgs.grh**

bld/gui/win/nt386/guimsgs.grh is missing

```
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2> ls bld/gui/win/nt386/guimsgs.grh
ls: cannot access 'bld/gui/win/nt386/guimsgs.grh': No such file or directory
linux@DESKTOP-G8ILNGQ:~/temp/open-watcom-v2_dev/open-watcom-v2> ls bld/gui/win/nt386
makefile
```

**Comment by jmalak** [comment]
Check gui project build log if in bld/gui/win/nt386 is builded objects.

**Comment by LowLevelMahn** [comment]
> Check gui project build log if in bld/gui/win/nt386 is builded objects.

i can't find any reference of `win/nt386` or similar patterns in `bld/gui/build.lo1`

but many to the other existing `ui/nt386`

**Comment by jmalak** [comment]
OK, it looks like GUI build is suppressed.
Check in your configuration script if you don't use following environment variables (should be commented out)
# export OWNOWGML=1
# export OWGUINOBUILD=1
# export OWNOBUILD=
if they are not commented then give me your setup.

**Comment by LowLevelMahn** [comment]
> OK, it looks like GUI build is suppressed. Check in your configuration script if you don't use following environment variables (should be commented out)
> 
> ```
> # export OWNOWGML=1
> # export OWGUINOBUILD=1
> # export OWNOBUILD=
> ```
> 
> if they are not commented then give me your setup.

you right ;/

![image](https://github.com/user-attachments/assets/96cbf435-8a00-479d-9294-6a461cb1d603)

i tested too much and didn't started freshly

**Comment by jmalak** [comment]
I see, the issue is with OWGUINOBUILD=1.
I will fix it because in aui project is incorrectly build GUI version even if it is suppressed by OWGUINOBUILD=1.

**Comment by jmalak** [comment]
Please try to rebuild with existing setup with corrected make files.

**Comment by LowLevelMahn** [comment]
> Please try to rebuild with existing setup with corrected make files.

already started with the minimal setvars - will do after with the setvars.sh version that get the errors

**Comment by jmalak** [comment]
If you removed OWGUINOBUILD=1 then it will be build all GUI tools that this issue cannot happen.
Now build with OWGUINOBUILD=1 should works correctly.

**Comment by LowLevelMahn** [comment]
still problems - because Tumblewee (also Fedora) uses dosbox-staging as default - dosbox aborts somewhere in between - need first ask the dosbox-staging guys to fix that - will test it later then again

**Comment by jmalak** [comment]
OK. If it need some special setup we can do it similar for OSX which need the changes.

**Comment by LowLevelMahn** [comment]
Issue on dosbox-staging: https://github.com/dosbox-staging/dosbox-staging/issues/3984

**Comment by jmalak** [comment]
Anyway you can try it with OWNOWGML=1 and all remaining setup as default.
It should suppress build only if it needs WGML tool (documentation and some tools) that it should build most of OW build including GUI tools.

**Comment by LowLevelMahn** [comment]
> Anyway you can try it with OWNOWGML=1 and all remaining setup as default. It should suppress build only if it needs WGML tool (documentation and a few tools) that it should build most of OW build including GUI tools.

i'll try that and give feedback

so open bugs:

Tumbleweed: the c99 bug and the problem with the problem default dosbox-staging
Fedora: only the problem default dosbox-staging

nothing you can fix

**Comment by jmalak** [comment]
We have temporary workaround for both.
c99 bug - change build script
DOSBOX - suppress use of wgml at all

**Comment by LowLevelMahn** [comment]
regarding the dosbox-staging fail

is it possible to just build that part so the dosbox-staging devs can only build that for testing?

**Comment by LowLevelMahn** [comment]
OWNOWGML=1 works on Tumbleweed - not dosbox gets called - not build error

**Comment by jmalak** [comment]
Yes, it is possible.
bellow is minimal steps to reproduce call to DOSBOX with minimal build.

Build minimal build system from scratch on Linux
It builds all tools for test sample, not need repeat if already done
- configure OW build environments for build host
- run **`sh build.sh preboot`**
- **`cd bld/watcom`**
- run **`builder boot`**
- **`cd ../bmp2eps`**
- run **`builder boot`**
- **`cd ../..`**

reproduce call to DOSBOX by
- **`cd docs/ps`**
- run **`wmake hbook=c_readme`**

if need repeat then cleanup docs/ps by
- run **`wmake clean`**

then you repeat call to DOSBOX by commands above

**Comment by LowLevelMahn** [comment]
> Yes, it is possible.
> bellow is minimal steps to reproduce call to DOSBOX with minimal build.

works!

any idea how to see exactly the call to dosbox - to get a minimal example

**Comment by jmalak** [comment]
setup OWWGMLDEBUG=1 it should show DOSBOX Window

anyway I found out simpler example

Build minimal build system from scratch on Linux
It builds a few tools necessary for test sample, not need repeat if already done
- configure OW build environments for build host
- run **`sh build.sh preboot`**

reproduce call to DOSBOX by
- **`cd docs/ps`**
- run **`wmake hbook=cgdoc`**

if need repeat then cleanup docs/ps by
- run **`wmake clean`**

then you can repeat call to DOSBOX by wmake command above

**Comment by LowLevelMahn** [comment]
i have found it

you set `set SDL_VIDEODRIVER=dummy` before calling dosbox

in `open-watcom-v2/build/mif/wgmlcmd.mif` line 75

that don't like staging

**Comment by jmalak** [comment]
Sorry I forgot how to see commands.

Change command by adding verbose value

**`wmake hbook=cgdoc verbose=1`**

**Comment by jmalak** [comment]
SDL_VIDEODRIVER=dummy
is legal value and works before on any platform

**Comment by LowLevelMahn** [comment]
> SDL_VIDEODRIVER=dummy
> is legal value and works before on any platform

i know - maybe its due to dosbox staging switch to SDL2 instead of SDL1 with dosbox vanilla

**Comment by jmalak** [comment]
offscreen is not driver, but init attribute
You ned to comment out both set SDL_VIDEODRIVER=dummy lines

**Comment by LowLevelMahn** [comment]
i changed the wgmlcmd.mif line with =dummy into =offscreen - but that gets someway ignored?
![image](https://github.com/user-attachments/assets/52da6cc4-7569-45d6-b09c-266cc2547203)

then i did a `./clean.sh` restarted the console an did

`. ./my_setvars.sh`
`./build.sh preboot`
`cd docs/ps`
`wmake hbook=cgdoc verbose=1`

still prints `set SDL_VIDEODRIVER=dummy`

![image](https://github.com/user-attachments/assets/90ee2f59-49e4-4fa8-b419-90e906fc881a)

**Comment by jmalak** [comment]
you need to run wmake clean in docs/ps to remove batch files

**Comment by LowLevelMahn** [comment]
> be careful there are 2 definitions you need to change both

found "both" 

works with SDL_VIDEODRIVER=offscreen and commenting them out

**Comment by LowLevelMahn** [comment]
`SDL_VIDEODRIVER=offscreen` works also for Ubuntu/DOSBox version 0.74-3

**Comment by jmalak** [comment]
SDL_VIDEODRIVER=offscreen doesn't work for "pure" old DOSBOX
I get 

	dosbox -conf "/home/jiri/ow2a/build/dosbox.cfg"  -c "mount c \"/home/jiri/ow2a\"" -c "c:\docs\ps\wgmlb.bat"
DOSBox version 0.74-3
Copyright 2002-2019 DOSBox Team, published under GNU GPL.
---
Exit to error: Can't init SDL No available video device

Error(E42): Last command making (run_wgml;cgdoc.ps) returned a bad status
Error(E02): Make execution terminated

**Comment by LowLevelMahn** [comment]
it depends on the SDL1 ability - maybe its not build with offscreen support

i try to find out what is compatible with SDL1 and SDL2

**Comment by jmalak** [comment]
I submit change fe858dfc to try to use default video setup instead of dummy

**Comment by LowLevelMahn** [comment]
that works - with poping up windows - but ok so far

**Comment by jmalak** [comment]
I am sorry but I must return DOSBOX setup back because we need to run DOSBOX on background without terminal.
It is reason for SDL_VIDEODRIVER=dummy that DOSBOX not need terminal any way.
As soon as some driver is used then it requires interactive session that doesn't exists.

If you want to use DOSBOX staging then you need to ask to fix it to work properly without terminal with SDL_VIDEODRIVER=dummy. It looks like they destroy this functionality.
We support standard DOSBOX 0.7x versions which are stable and support this mode correctly.

**Comment by LowLevelMahn** [comment]
> I am sorry but I must return DOSBOX setup back because we need to run DOSBOX on background without terminal.

no problem - having poping up windows all the while is stressy or even not possible when not gfx is availabe

**Comment by LowLevelMahn** [comment]
thanks for all the help - i will add another Tumbleweed part and Windows builds part to the wiki - with all the infos i got now
